db/blob/blob_file_reader.h

```diff
@@ -0,0 +1,81 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <cinttypes>
+#include <memory>
+
+#include "file/random_access_file_reader.h"
+#include "rocksdb/compression_type.h"
+#include "rocksdb/rocksdb_namespace.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class Status;
+struct ImmutableCFOptions;
+struct FileOptions;
+class HistogramImpl;
+struct ReadOptions;
+class Slice;
+class PinnableSlice;
+
+class BlobFileReader {
```

Comment:
`BlobLogReader` is legacy code from the old implementation. It is currently only used in tests (technically, it is also referenced by some dead code in the old BlobDB that I plan to rip out) but I think we could also utilize it in the `blob_dump` tool (which is exactly the use case you suggested).

Reply:
Then I suggest that we give better names to the two classes.

Reply:
I really like `BlobFileBuilder` / `BlobFileReader` because they rhyme with `TableBuilder` / `TableReader`. We could rename `BlobLogReader` to something like `BlobLogSequentialReader` though if you think.

Reply:
Let me put out a separate PR for the rename; it's only tangentially related and this is already a pretty big patch.